src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  availableAmount,
+  canAdventure,
+  currentRound,
+  inebrietyLimit,
+  myAdventures,
+  myInebriety,
+  myLevel,
+  runChoice,
+  runCombat,
+  totalTurnsPlayed,
+  toUrl,
+  use,
+  useFamiliar,
+  visitUrl,
+} from "kolmafia";
+import {
+  $effect,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $skill,
+  adventureMacro,
+  adventureMacroAuto,
+  clamp,
+  Counter,
+  ensureEffect,
+  get,
+  getModifier,
+  have,
+  Requirement,
+  SourceTerminal,
+} from "libram";
+import { Macro, withMacro } from "./combat";
+import { completeBarfQuest } from "./dailies";
+import { estimatedTurns } from "./embezzler";
+import { barfFamiliar, freeFightFamiliar, meatFamiliar } from "./familiar";
+import {
+  embezzlerLog,
+  globalOptions,
+  kramcoGuaranteed,
+  safeRestore,
+  setChoice,
+  steveAdventures,
+} from "./lib";
+import { meatMood } from "./mood";
+import {
+  familiarWaterBreathingEquipment,
+  freeFightOutfit,
+  meatOutfit,
+  tryFillLatte,
+  waterBreathingEquipment,
+} from "./outfit";
+import { determineDraggableZoneAndEnsureAccess, digitizedMonstersRemaining } from "./wanderer";
+const embezzler = $monster`Knob Goblin Embezzler`;
+
+function embezzlerPrep(requirements?: Requirement) {
+  useFamiliar(meatFamiliar());
+  meatOutfit(true, requirements);
+}
+
+function freeFightPrep(requirements?: Requirement) {
+  useFamiliar(freeFightFamiliar(true));
+  freeFightOutfit(requirements);
+}
+
+const sober = () => myInebriety() <= inebrietyLimit();
+
+type Turn = {
+  name: string;
+  available: () => boolean;
+  perform: () => boolean;
+};
+
+const turns: Turn[] = [
+  {
+    name: "Lights Out",
+    available: () =>
+      totalTurnsPlayed() % 37 === 0 && totalTurnsPlayed() !== get("lastLightsOutTurn"),
+    perform: () => {
+      const steveRoom = get("nextSpookyravenStephenRoom");
+      const ghostLocation = get("ghostLocation");
+      if (steveRoom && canAdventure(steveRoom) && steveRoom !== ghostLocation) {
+        const fightingSteve = steveRoom === $location`The Haunted Laboratory`;
+        // Technically drops 500 meat, but that's close enough for me.
+        if (fightingSteve) embezzlerPrep();
+        const plan = steveAdventures.get(steveRoom);
+        if (plan) {
+          withMacro(
+            Macro.if_($monster`Stephen Spookyraven`, Macro.basicCombat()).abort(),
+            () => {
+              visitUrl(toUrl(steveRoom));
+              for (const choiceValue of plan) {
+                runChoice(choiceValue);
+              }
+              if (fightingSteve || currentRound()) runCombat();
+            },
+            true
+          );
+        }
+        return fightingSteve;
+      }
+      return false;
+    },
+  },
+  {
+    name: "Proton Ghost",
+    available: () =>
+      sober() &&
```

Comment:
i don't get why we are doing the sober check here. even if you can't shoot ghost, you should be able to still kill it if you're running while overdrunk w/ the wine glass. you can't ghostbust but it's still a free fight to increment tiny bits and bobs of value.

Reply:
We have historically always checked for sobriety because some people have trouble killing physically resistant monsters with only weapon attacks. It's worth re-investigating.